src/test/java/com/bookshop/service/CategoryServiceTests.java

```diff
@@ -0,0 +1,289 @@
+package com.bookshop.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import com.bookshop.dto.category.request.CategoryDtoRequest;
+import com.bookshop.dto.category.response.CategoryDto;
+import com.bookshop.exception.EntityNotFoundException;
+import com.bookshop.mapper.CategoryMapper;
+import com.bookshop.model.Category;
+import com.bookshop.repository.category.CategoryRepository;
+import com.bookshop.service.impl.CategoryServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceTests {
+    private static final Long ID_ONE = 1L;
+    private static final Long INCORRECT_ID = 100L;
+    private static final int ONCE = 1;
+    private static final Pageable STANDART_PAGEABLE = PageRequest.of(0, 20);
+
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @Test
+    @DisplayName("Save category by correct data")
+    public void save_AllCorrectData_CorrectDtoReturned() {
+        //given
+        CategoryDtoRequest requestDto = createValidCategoryDtoRequest();
+        Category expectedCategoryWithoutId = getCategoryFromCreateCategoryRequestDto(requestDto);
+        Category expectedCategoryWithId = getCategoryFromCreateCategoryRequestDto(requestDto);
+        expectedCategoryWithId.setId(ID_ONE);
+        CategoryDto expected = getCategoryDtoFromCategory(expectedCategoryWithId);
+
+        when(categoryMapper.toCategory(requestDto)).thenReturn(expectedCategoryWithoutId);
+        when(categoryRepository.save(expectedCategoryWithoutId)).thenReturn(expectedCategoryWithId);
+        when(categoryMapper.toDto(expectedCategoryWithId)).thenReturn(expected);
+
+        //when
+        CategoryDto actual = categoryService.save(requestDto);
+
+        //then
+        Assertions.assertEquals(expected, actual);
+        verify(categoryRepository, times(ONCE)).save(expectedCategoryWithoutId);
```

Comment:
```suggestion
        verify(categoryRepository).save(expectedCategoryWithoutId);
```
`times` is not required when you expect 1 as it is a default behaviour when you don't specify anything, but decide what is better for you